File: c++/타겟-넘버.cpp
#include <string>
#include <vector>

using namespace std;
int dfs(vector<int> numbers, int target, int index, int sum){
	// 종료 조건 : 모두 탐색한 경우 
    if (numbers.size() == index){
        return sum == target ? 1 : 0;
    }
    // sum에 더하기 연산을 한 경우와 빼기 연산을 한 경우를 합함
    else {
        return dfs(numbers, target, index + 1, sum + numbers[index]) + dfs(numbers, target, index + 1, sum - numbers[index]);
    }
}

int solution(vector<int> numbers, int target) {
    return dfs(numbers, target, 0, 0);
}

File: python/타겟-넘버.py
def solution(numbers, target):
    return dfs(numbers, target, 0, 0)

def dfs(numbers, target, sum, index):
    if len(numbers) == index:
        return 1 if sum == target else 0
    else:
        return dfs(numbers, target, sum + numbers[index], index + 1) + dfs(numbers, target, sum - numbers[index], index + 1)
